session.go

```diff
@@ -561,7 +565,7 @@ func (s *session) DropPreparedStmt(stmtID uint32) error {
 	if _, ok := vars.PreparedStmts[stmtID]; !ok {
 		return executor.ErrStmtNotFound
 	}
-	delete(vars.PreparedStmts, stmtID)
+	vars.RetryInfo.DroppedPreparedStmtIDs = append(vars.RetryInfo.DroppedPreparedStmtIDs, stmtID)
```

Comment:
In retry, `ExecutePreparedStmt` will execute with the same stmtID, so we should not dropping the preparedStmts until finished the transaction, dropping it in ClearRetryInfo.